src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs

```diff
@@ -46,15 +47,25 @@ public DiagnosticsMiddleware(OwinMiddleware next)
     /// <inheritdoc />
     public override async Task Invoke(IOwinContext owinContext)
     {
+        Stopwatch stopwatch = null;
+
         try
         {
             BeginRequest(owinContext);
+
+            if (OwinInstrumentationMetrics.HttpClientDuration.Enabled && !owinContext.Environment.ContainsKey(ContextKey))
+            {
+                stopwatch = Stopwatch.StartNew();
```

Comment:
Stopwatch has a static [GetTimestamp](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch.gettimestamp?view=netframework-4.8) method which returns the raw value. If we capture that value at the start and then at the end could we do some math to calculate the duration and avoid the allocation cost? It looks like the class is really just doing that under the hood: https://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Stopwatch.cs,87